### PR TITLE
Basic room generator, enable player collision

### PR DESCRIPTION
**Pull request by jeffsieu**
Adds a LevelGenerator class which can generate a rectangular room of a given size, with four walls of a specified thickness and height.
image

### CODE REVIEW

**Comment by jeffsieu** [comment]
Not sure if this is the optimal way to organize the functions, in terms of the arguments and stuff.

File: Scripts/LevelGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        // Also, we need to offset the wall's center to the side by half the wall's thickness.
+        PhysicsBody frontWall = GenerateWall(origin + ((roomSize.y - wallThickness) / 2) * front + (wallThickness / 2) * left, frontBackWallSize, -front);
+        PhysicsBody backWall = GenerateWall(origin - ((roomSize.y - wallThickness) / 2) * front - (wallThickness / 2) * left, frontBackWallSize, front);
+        PhysicsBody leftWall = GenerateWall(origin + ((roomSize.x - wallThickness) / 2) * left - (wallThickness / 2) * front, leftRightWallSize, -left);
+        PhysicsBody rightWall = GenerateWall(origin - ((roomSize.x - wallThickness) / 2) * left + (wallThickness / 2) * front, leftRightWallSize, left);
+
+        AddChild(frontWall);
+        AddChild(backWall);
+        AddChild(leftWall);
+        AddChild(rightWall);
```
Comment (daniellimws) [inline Scripts/LevelGenerator.cs]: Need to keep references to them so that we can destroy them later?

Comment (jeffsieu) [reply Scripts/LevelGenerator.cs]: Can probably add that in a later part I guess, where we can optimise by removing and adding depending on whether it's visible.

**Review by daniellimws — APPROVED** [review]
I think is ok. Move this into a folder Generator/ then can merge.

File: Scripts/LevelGenerator.cs (hunk 1 above)
Comment (owenl131) [reply Scripts/LevelGenerator.cs]: Maybe can return it in the return value.

**Pull request merged**